pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -169,66 +176,178 @@ pub mod pallet {
 	impl<T: Config> Pallet<T> {
 		#[pallet::call_index(0)]
 		#[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
-		pub fn create_pool(_origin: OriginFor<T>) -> DispatchResult {
-			todo!()
+		pub fn create_pool(
+			origin: OriginFor<T>,
+			curve: CurveInput<CurveParameterInputOf<T>>,
+			currencies: BoundedVec<TokenMetaOf<T>, T::MaxCurrencies>,
+			denomination: u8,
+			pool_manager: AccountIdOf<T>,
+			transferable: bool,
+		) -> DispatchResult {
+			let who = T::PoolCreateOrigin::ensure_origin(origin)?;
+
+			let currency_length = currencies.len();
+
+			let checked_curve = curve.try_into().map_err(|_| Error::<T>::InvalidInput)?;
+
+			let current_asset_id = NextAssetId::<T>::get();
+
+			let (currency_ids, next_asset_id) = Self::generate_sequential_asset_ids(current_asset_id, currency_length)?;
+
+			let pool_id = T::PoolId::from(currency_ids.blake2_256());
+
+			// Todo: change that.
+			T::DepositCurrency::hold(
+				&T::RuntimeHoldReason::from(HoldReason::Deposit),
+				&who,
+				T::BaseDeposit::get().saturating_add(
+					T::DepositPerCurrency::get()
+						.saturating_mul(currency_length.saturated_into())
+						.saturated_into(),
+				),
+			)?;
+
+			currencies
+				.iter()
+				.enumerate()
+				.try_for_each(|(idx, entry)| -> DispatchResult {
+					let asset_id: &T::AssetId = currency_ids.get(idx).ok_or(Error::<T>::IndexOutOfBounds)?;
+					T::Fungibles::create(asset_id.clone(), pool_id.clone().into(), false, entry.min_balance)?;
+
+					// set metadata for new asset class
+					T::Fungibles::set(
+						asset_id.clone(),
+						&pool_id.clone().into(),
+						entry.name.clone().into(),
+						entry.symbol.clone().into(),
+						denomination,
+					)?;
+
+					Ok(())
+				})?;
+
+			// Touch the pool account in order to be able to transfer the collateral currency to it
+			T::CollateralCurrency::touch(T::CollateralAssetId::get(), &pool_id.clone().into(), &who)?;
+
+			Pools::<T>::set(
+				&pool_id,
+				Some(PoolDetails::new(
+					pool_manager,
```

Comment:
btw I think we need a manager and an owner, bc we want to allow 'unmanaged' pools (nobody has special permission) but still we need an owner to send and excess collateral to - and we may still grant the owner the right to destroy a pool.

Reply:
Made a proposal for that on #772

Reply:
I will also wait tackling that until the proposal is merged. Some for: https://github.com/KILTprotocol/kilt-node/pull/768#discussion_r1820572239